crates/core/src/factories/core_space.rs

```diff
@@ -92,13 +134,142 @@ impl Space for CoreSpace {
+                            info,
+                        )
+                        .await
+                        {
+                            Err(err) => {
+                                tracing::warn!(
+                                    ?err,
+                                    "failed to sign agent info",
+                                );
+                                return;
+                            }
+                            Ok(info) => info,
+                        };
+
+                        // add it to the peer_store.
+                        if let Err(err) = peer_store.insert(vec![info]).await {
+                            tracing::warn!(
+                                ?err,
+                                "failed to add agent info to peer store"
+                            );
+                        }
+
+                        // TODO - send the new agent info to bootstrap.
+                    }
+                });
+            }));
+
+            // trigger the update
+            local_agent.invoke_cb();
+
+            Ok(())
+        })
     }
 
-    fn local_agent_leave(&self, _local_agent: id::AgentId) -> BoxFut<'_, ()> {
-        Box::pin(async move { todo!() })
+    fn local_agent_leave(&self, local_agent: id::AgentId) -> BoxFut<'_, ()> {
+        Box::pin(async move {
+            // TODO - inform gossip module of leave
+            // TODO - inform sharding module of leave
+
+            let local_agent = self
+                .inner
+                .lock()
+                .unwrap()
+                .local_agent_map
+                .remove(&local_agent);
+
+            if let Some(local_agent) = local_agent {
+                // register a dummy update cb
+                local_agent.register_cb(Arc::new(|| ()));
+
+                // sign a new tombstone
+                let created_at = Timestamp::now();
+                let expires_at =
+                    created_at + std::time::Duration::from_secs(60 * 20);
+                let info = agent::AgentInfo {
+                    agent: local_agent.agent().clone(),
+                    space: self.space.clone(),
+                    created_at,
+                    expires_at,
+                    is_tombstone: true,
+                    url: None,
+                    storage_arc: DhtArc::Empty,
+                };
+
+                let info = match agent::AgentInfoSigned::sign(
+                    &local_agent,
+                    info,
+                )
+                .await
+                {
+                    Err(err) => {
+                        tracing::warn!(?err, "failed to sign agent info");
+                        return;
+                    }
+                    Ok(info) => info,
+                };
+
+                if let Err(err) = self.peer_store.insert(vec![info]).await {
+                    tracing::warn!(
+                        ?err,
+                        "failed to add agent info to peer store"
```

Comment:
The method name is shown in the trace as well, I believe, but this describes the operation a bit clearer
```suggestion
                        "failed to tombstone agent info in peer store"
```